File: FrontendOfSpringProject/src/app/models/product-offer-dto.ts
export class ProductOfferDTO {
    id?: number;
    productCode!: string;
    offerCode!: string;
    offerDetails?: OfferDetailsDTO;
    offerStartDate!: string;
    offerEndDate!: string;
  }
  
  export class OfferDetailsDTO {
    id?: number;
    code!: string;
    description: string | undefined;
    value?: string;
    status?: string;
    activationDate?: string;
    expiryDate?: string;
  }
  

File: ProductManagementSystem2/src/main/java/com/productManagement/dto/ProductOfferDTO.java
package com.productManagement.dto;

import java.time.LocalDate;

import lombok.Data;

@Data
public class ProductOfferDTO {

	private Long id;
	private String productCode;
	private String offerCode;
	private OfferDetailsDTO offerDetails; // Complete offer details
//    private ProductDetailsDTO productDetails;
	private LocalDate offerStartDate;
	private LocalDate offerEndDate;

}
